packages/mui-joy/src/FormControl/FormControl.test.tsx

```diff
@@ -110,6 +110,24 @@ describe('<FormControl />', () => {
 
       expect(getByRole('textbox')).to.have.attribute('required');
     });
+
+    it('should inherit htmlFor from FormControl if htmlFor is undefined', () => {
+      const { container } = render(
+        <FormControl>
+          <FormLabel htmlFor={undefined} />
```

Comment:
Let's add label; that would be a more practical example:
```suggestion
          <FormLabel htmlFor={undefined}>Label</FormLabel>
```